src/ztex/task.c

```diff
@@ -373,23 +418,49 @@ void task_result_execute(struct task_list *task_list,
 	struct task *task;
 	for (task = task_list->task; task; task = task->next) {
 		struct task_result *result;
-		for (result = task->result_list; result; result = result->next)
+		for (result = task->result_list.result_list; result;
+				result = result->next)
 			func(result);
 	}
 }
 
 
+void task_list_create_index(struct task_list *task_list)
+{
+	struct task *task;
+	for (task = task_list->task; task; task = task->next)
+		task_result_list_create_index(&task->result_list);
+}

```

Comment:
Is this function/loop a no-op (never calls `task_result_list_create_index`) when `crypt_all` would return 0 (no matches)? I hope so, I just want this discussed/confirmed explicitly.

Reply:
I think it would look better to skip indexing operations when ```crypt_all()``` would return 0, because on the subsequent ```crypt_all()``` it drops existing ```task_list``` (with ```task_result```'s and their indices) and creates new ```task_list``` (with new portion of data).